src/client/datascience/notebook/notebookControllerManager.ts

```diff
@@ -200,6 +204,38 @@ export class NotebookControllerManager implements INotebookControllerManager, IE
         }
         return this.getOrCreateController(activeInterpreter, notebookType);
     }
+    private async createDefaultRemoteControllerForInteractiveWindow() {
+        // Get all remote kernels
+        await this.loadNotebookControllers();
+        const controllers = this.registeredNotebookControllers();
+        if (controllers.length === 0) {
+            traceError('No remote controllers');
+            return;
+        }
+
+        // Find the default kernel `python` if we can find one
+        // If not available, then return anything thats a python kernel
+        let defaultPythonKernel: VSCodeNotebookController | undefined;
+        let defaultPython3Kernel: VSCodeNotebookController | undefined;
+        let defaultPythonLanguageKernel: VSCodeNotebookController | undefined;
+        controllers.forEach((item) => {
+            if (item.connection.kind === 'startUsingKernelSpec' && item.connection.kernelSpec.name === 'python') {
+                defaultPythonKernel = item;
+            } else if (
+                item.connection.kind === 'startUsingKernelSpec' &&
+                item.connection.kernelSpec.name === 'python3'
+            ) {
+                defaultPython3Kernel = item;
+            } else if (
+                item.connection.kind === 'startUsingKernelSpec' &&
+                item.connection.kernelSpec.language === PYTHON_LANGUAGE
+            ) {
+                defaultPythonLanguageKernel = item;
+            }
+        });
+
+        return defaultPythonKernel || defaultPython3Kernel || defaultPythonLanguageKernel || controllers[0];
```

Comment:
For the default kernel would we maybe want defaultPython3Kernel to have priority over defaultPythonKernel? Feel like that would be better on mac / linux systems.

Reply:
@rchiodo @IanMatthewHuff i'll be updating this code, as this is assuming IW always works with python, and I do believe we used to support other languages.

Reply:
It used to support other languages from the kernel picker AFAIK, but not on startup.